WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java

```diff
@@ -696,4 +705,15 @@ public void onMediaChanged(MediaStore.OnMediaChanged event) {
             }
         }
     }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onMediaUploaded(OnMediaUploaded event) {
+        if (event.media != null && event.completed) {
+            PostModel post = mPostStore.getPostByLocalPostId(event.media.getLocalPostId());
+            if (post != null && PostUploadService.isPostUploadingOrQueued(post)) {
+                loadPosts(LoadMode.FORCED);
```

Comment:
nice addition here - refresh post list when a new post has just been uploaded. Do you know if this code is run as well when some other activity is in the foreground, i.e. the Post Preview or the Editor for a post other than the one that has just been finished uploading ?